packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -975,15 +978,24 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this.updateDocumentDirtyState(newState);
     }
 
-    public setConnectionState(connected: boolean, clientId?: string) {
+    public async setConnectionState(connected: boolean, clientId?: string) {
         this.verifyNotClosed();
 
         // There might be no change of state due to Container calling this API after loading runtime.
         const changeOfState = this._connected !== connected;
         this._connected = connected;
 
         if (changeOfState) {
-           this.replayPendingStates();
+            if (this.context.pendingOps !== undefined) {
+                const pendingOps = this.pendingStateManager.getPendingMessages();
+                if (pendingOps.length > 0) {
+                    await Promise.all(pendingOps.map(async (op) => {
+                        return this.dataStores.loadChannelFromOp(op);
```

Comment:
Not sure I follow the purpose of this code. From what I see, we use content of the ops to load relevant objects into memory only. By the time we get "connected" event, all objects should have loaded already and processed local ops, so this should be noop, I think

Reply:
Hm, looks like things might not be loaded at the end of the process, as we put "rebase" ops into this.pending.
I think I'd rather see rebaseOp asynchronous and forcing load of particular channel and processing op. That seems more natural to me than having code in here to do the same.